src/browser/api/service.ts

```diff
@@ -50,18 +51,20 @@ export module Service {
     export function registerService (identity: Identity, serviceName?: string): ServiceIdentity | false {
         const serviceIdentity = getExternalOrOfWindowIdentity(identity);
         // If a service is already registered from that uuid, nack
-        if (!serviceIdentity || serviceMap.get(serviceIdentity.uuid)) {
+        if (!serviceIdentity || getServiceByUuid(identity.uuid)) {
             return false;
         }
         serviceIdentity.serviceName = serviceName;
```

Comment:
Here I see a variable `serviceIdentity` which by convention, I would assume is of Interface type `ServiceIdentity`... (particularly because it is subsequently added to `serviceMap` of type `<string, ServiceIdentity>`.

Assigning `serviceName` should not be allowed here.